docs/configuration/extract-codacy-logs.sh

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env bash
+
+#### Helper functions ####
+
+usage()
+{
+    echo "Usage: $0 -n <NAMESPACE> "
+    exit 3
+}
+
+set_variable()
+{
+    local varname=$1
+    shift
+    if [ -z "${!varname}" ]; then
+        eval "$varname=\"$@\""
+    else
+        echo "Error: $varname already set"
+        usage
+    fi
+}
+
+cleanup()
+{
+    if [ -d "$LOGS_DIR" ]; then
+        rm -r $LOGS_DIR &>/dev/null
+    fi
+}
+trap cleanup EXIT
+
+#### Log extraction script ####
+
+while getopts "n:" option; do
+    case ${option} in
+        n )
+            set_variable NAMESPACE ${OPTARG}
+            ;;
+        \? )
+            usage
+            ;;
+    esac
+done
+
+[ -z "$NAMESPACE" ] && usage
+
+CURRENT_DATE_TIME=$(date "+%Y%m%d-%H%M%S")
+LOGS_DIR="$(mktemp -d)/codacy_logs_$CURRENT_DATE_TIME"
+
+echo "Starting log files extraction"
+
+# Check if kubectl is available
+echo "Checking if kubectl is installed..."
+if ! kubectl version &>/dev/null; then
```

Comment:
In that case I need the value returned from the command... I guess I could use `$?` if no other commands are issued in between.

Reply:
I can't because `$?` has the exit code and not the command output. Maybe I could do it in a smarter way, but the way it is right now is simple enough and works.